src/Admin/Pool.php

```diff
@@ -239,6 +239,14 @@ public function hasAdminByClass($class)
      */
     public function getAdminByAdminCode($adminCode)
     {
+        if (!\is_string($adminCode)) {
+            @trigger_error(sprintf(
+                'Passing an non string value as argument 1 for %s() is deprecated since 3.x and will throw an exception in 4.0.',
```

Comment:
```suggestion
                'Passing a non string value as argument 1 for %s() is deprecated since sonata-project/admin-bundle 3.x and will throw an exception in 4.0.',
```